### DISCUSSION TITLE
How to create a rectangle Waveform? #69

### DISCUSSION THREAD

**Discussion by icegh** [naudio/NAudio#961] [Q&A] [answered]
dear community,

i am new in c# and also using first time NWaves lib.

Can anybody helps me how to create a rectangle like that (i have attached a image).
i have no idead which type of signal builder i should use. has already tried all but no chance to get the same results.

In audacity i can simple create this via menu. But how i can do that with the lib?
![14-11-_2022_20-33-34](https://user-images.githubusercontent.com/100025138/201752230-467146da-6e4c-496b-8bbf-a7809255264a.jpg)

thank you so much

**Comment by markheath** [comment, marked as answer]
This is called a square wave, and you can use the NAudio `SignalGenerator` class to generate one

**Reply by icegh** [reply]
Okay fine, thank you so much. I am completely new in DSP. Thank you
